internal/btf/btf.go

```diff
@@ -65,10 +65,25 @@ func (h *btfHeader) stringStart() int64 {
 	return int64(h.HdrLen + h.StringOff)
 }
 
-// LoadSpecFromReader reads BTF sections from an ELF.
+// LoadSpecFromReader reads from an ELF or a raw BTF blob.
 //
-// Returns ErrNotFound if the reader contains no BTF.
+// Returns ErrNotFound if reading from an ELF which contains no BTF.
 func LoadSpecFromReader(rd io.ReaderAt) (*Spec, error) {
```

Comment:
I think the control flow could be cleaned up here. I don't think we really need to check for ELF magic, `NewSafeELFFile` already does that.

Instead, `guessBTFByteOrder()` could be called on the happy path as a first step and do an early return of `loadRawSpec()` if BTF magic is found. Otherwise, fall through to parsing it as an ELF, and the existing code further down will complain if anything's wrong.

Reply:
I ended up reshuffling this so that we fall back to raw BTF if opening the elf file fails for some reason. This means that ELF stays the happy path.

Reply:
Sounds good :+1: